thavalon-webapp/src/components/formButton.tsx

```diff
@@ -0,0 +1,48 @@
+// This module contains code related to the FormButton componenet, used for
+// submitting forms that might require loading.
+
+import React, { useState } from 'react';
+import './formButton.scss';
+
+/**
+ * Properties for the form button.
+ * @param label  The label the button will use
+ * @param isLoading  Whether or not the button should be in a loading state
+ * @param color  The color of the button. Defaults to green.
+ * @param size  The size of the button. Defaults to medium.
+ */
+export interface FormButtonProps {
+    label: string
+    isLoading: boolean
+    color?: "red" | "green" | "grey"
+    size?: "small" | "medium" | "large"
+}
+
+/**
+ * The React component that renders the button and handles animations
+ * @param props The properties used by the FormButton. 
+ * @returns  The JSX.Element for the button.
+ */
+export function FormButton(props: FormButtonProps): JSX.Element {
+    // Handle loading state. If we should be loading, set a timeout to add the 
+    // `loader` class which spins. 
+    const [loader, setLoader] = useState("");
+    if (props.isLoading === true && loader === "") {
+        setTimeout(() => setLoader("loader"), 150);
+    }
+
+    // Add in classes. Only add "active" and "loader" if isLoading is true, since
+    // these cause animations.
+    const classes = `form-button-${ props.color || "green" } ` + (props.isLoading ? `active ${ loader }` : "");
```

Comment:
I think `className` can also take an array, that might be clearer to compose here